src/vivarium/framework/engine.py

```diff
@@ -57,6 +59,7 @@ def setup(self):
         self.time_step_emitters = {k: self.builder.event.get_emitter(k) for k in self.time_step_events}
         self.end_emitter = self.builder.event.get_emitter('simulation_end')
         self.builder.event.get_emitter('post_setup')(None)
+        self._setup = True
```

Comment:
Move this to before you emit post setup so components don't try to do anything weird during that lifecycle phase.